superset-frontend/src/dashboard/components/Header/index.jsx

```diff
@@ -129,7 +129,10 @@ const StyledDashboardHeader = styled.div`
 
 class Header extends React.PureComponent {
   static discardChanges() {
-    window.location.reload();
+    const url = new URL(window.location.href);
+
+    url.searchParams.delete('edit');
+    window.location.replace(url);
```

Comment:
We should use react router for this instead, if possible.

Reply:
I agree. However, the original implementation used window reload for simplicity, otherwise we would need to implement a method to remove ptogrammatically any component that was added to the dashboard and that wasn't saved yet. As this is a P1 bug, I think we should fix the original problem first and consider making a better implementation in a later stage.

Reply:
I'm also curious if there might be any expectation for the back button to work. If so, it might be worth trying `window.location.assign(url)` or `window.location = url` or `window.location.href = url` (which all do the same thing)

Reply:
@rusackas Changing to `assign` allows to go back to the edit mode indeed which is what you are looking for, I guess. If the expectation is to undo the "discard changes" action instead, the back button won't be able to achieve that. Every component added and then discarded will be gone when going back.